pkg/cmd/cli/cmd/process.go

```diff
@@ -111,122 +90,152 @@ func RunProcess(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []
 	}
 
 	var (
-		templateObj *api.Template
-		mapping     *meta.RESTMapping
+		objects []runtime.Object
+		mapping *meta.RESTMapping
 	)
 
-	if len(storedTemplate) > 0 {
-		templateObj, err = client.Templates(namespace).Get(storedTemplate)
-		if err != nil {
-			if errors.IsNotFound(err) {
-				return fmt.Errorf("template %q could not be found", storedTemplate)
-			}
-			return err
-		}
+	version, kind, err := mapper.VersionAndKindForResource("template")
+	if mapping, err = mapper.RESTMapping(kind, version); err != nil {
+		return err
+	}
 
-		version, kind, err := mapper.VersionAndKindForResource("template")
-		if mapping, err = mapper.RESTMapping(kind, version); err != nil {
-			return err
-		}
-	} else {
-		obj, err := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
+	outputVersion := kcmdutil.OutputVersion(cmd, mapping.APIVersion)
+	raw := kcmdutil.GetFlagBool(cmd, "raw")
+
+	var infos []*resource.Info
+
+	if len(storedTemplate) == 0 {
+		infos, err = resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
 			NamespaceParam(namespace).RequireNamespace().
 			FilenameParam(filename).
 			Do().
-			Object()
-
+			Infos()
 		if err != nil {
 			return err
 		}
-
-		var ok bool
-		templateObj, ok = obj.(*api.Template)
-		if !ok {
-			return fmt.Errorf("cannot convert input to Template: ", reflect.TypeOf(obj))
-		}
-		templateObj.CreationTimestamp = util.Now()
-
-		version, kind, err := kapi.Scheme.ObjectVersionAndKind(templateObj)
+	} else {
+		templateObj, err := client.Templates(namespace).Get(storedTemplate)
 		if err != nil {
+			if errors.IsNotFound(err) {
+				return fmt.Errorf("template %q could not be found", storedTemplate)
+			}
 			return err
 		}
-		if mapping, err = mapper.RESTMapping(kind, version); err != nil {
-			return err
-		}
+		templateObj.CreationTimestamp = util.Now()
+		infos = append(infos, &resource.Info{Object: templateObj})
 	}
 
-	if cmd.Flag("value").Changed {
-		injectUserVars(cmd, templateObj)
+	outputFormat := kcmdutil.GetFlagString(cmd, "output")
+	if len(outputFormat) == 0 {
+		outputFormat = "json"
 	}
 
-	// If 'parameters' flag is set it does not do processing but only print
-	// the template parameters to console for inspection.
-	if kcmdutil.GetFlagBool(cmd, "parameters") == true {
-		err = describe.PrintTemplateParameters(templateObj.Parameters, out)
-		if err != nil {
-			return err
+	for i := range infos {
+		obj, ok := infos[i].Object.(*api.Template)
```

Comment:
@kargakis i would rather iterate over existing array than working on the copy returned from info...